### PR TITLE
batches: Isolate ssh credentials oobmigration

### PR DESCRIPTION
**Pull request by efritz**
This PR _isolates_ oobmigration [#2](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@aed4b515082b4374f7d14fceeee98790c08a1361/-/blob/enterprise/cmd/worker/internal/batches/migrations/ssh_migrator.go) so that it does not need any _behavioral_ imports that can change over time. This is in preparation for its deprecation, where it will run only as part of the migrator for historic upgrades. We want to *freeze* this migration as much as possible, because we want to treat it with consistent behavior over time to allow the migrator to make some assumptions (like the registered code for an oobmigration is valid at any point in its lifetime).

This PR makes progress towards #38052, but still needs to be fed configuration keys from the worker. A follow-up PR will take care of this.

## Test plan

Existing unit tests.

### CODE REVIEW

File: enterprise/internal/oobmigration/migrations/batches/migrations.go [hunk 1]
```diff
@@ -3,44 +3,15 @@ package batches
...
-	cstore := store.New(db, observationContext, keyring.Default().BatchChangesCredentialKey)
-
-	// Register Batch Changes OOB migrations.
-	return Register(cstore, outOfBandMigrationRunner)
-}
-
-// Register registers all currently implemented out of band migrations
-// by batch changes with the migration runner.
-func Register(cstore *store.Store, outOfBandMigrationRunner *oobmigration.Runner) error {
+	sshMigrator := NewSSHMigratorWithDB(db, keyring.Default().BatchChangesCredentialKey)
```
Comment (efritz) [inline enterprise/internal/oobmigration/migrations/batches/migrations.go:12]: We need to figure out how to support encryption keys in the migrator still.

**Review by eseliger — APPROVED** [review]
shouldn't the tests be isolated, too?

**Comment by efritz** [comment]
> shouldn't the tests be isolated, too?

baby steps

**Comment by efritz** [comment]
@sourcegraph/batchers Would you like to deprecate this oobmigration? That would allow you to drop the column `ssh_migration_applied` from the `user_credentials_ table`, as well as the `BasicAuth` and `OAuthBearerToken` (without ssh) in your types packages.

**Comment by eseliger** [comment]
Yes, we would like to.

**Comment by efritz** [comment]
Follow-up PRs will include isolation of the tests and supplying encryption keys in a uniform way.

**Comment by efritz** [comment]
@eseliger  I've deprecated the oobmigrations in #40759. Feel free to add new ones anytime. For the ones that were deprecated, code on or after the 4.0 can assume 100% completion. Feel free to update the database schema/code to reflect that.

**Pull request merged**